src/Map/src/Icon/Icon.php

```diff
@@ -97,14 +112,15 @@ public function toArray(): array
             'type' => $this->type->value,
             'width' => $this->width,
             'height' => $this->height,
+            'color' => $this->color, // Ajout de la couleur au tableau de sérialisation
```

Comment:
The comment does not bring any value

```suggestion
            'color' => $this->color,
```